test/Chainweb/Test/BlockHeader/Validation.hs

```diff
@@ -157,58 +157,58 @@ prop_fail_validate = testCase "validate invalid BlockHeaders" $ do
-      , [IncorrectHash]
+      , [IncorrectHash, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockHash %~ messWords encodeBlockHash decodeBlockHash (flip complementBit 0)
-      , [IncorrectHash]
+      , [IncorrectHash, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockTarget %~ messWords encodeHashTarget decodeHashTarget (flip complementBit 0)
-      , [IncorrectHash, IncorrectPow, IncorrectTarget]
+      , [IncorrectHash, IncorrectPow, IncorrectTarget, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockParent %~ messWords encodeBlockHash decodeBlockHash (flip complementBit 0)
       , [MissingParent]
       )
     , ( hdr & testHeaderHdr . blockPayloadHash %~ messWords encodeBlockPayloadHash decodeBlockPayloadHash (flip complementBit 0)
-      , [IncorrectHash, IncorrectPow]
+      , [IncorrectHash, IncorrectPow, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockEpochStart %~ add second
-      , [IncorrectHash, IncorrectPow, IncorrectEpoch]
+      , [IncorrectHash, IncorrectPow, IncorrectEpoch, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockChainId .~ unsafeChainId 1
-      , [IncorrectHash, IncorrectPow, ChainMismatch, AdjacentChainMismatch]
+      , [IncorrectHash, IncorrectPow, ChainMismatch, AdjacentChainMismatch, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockChainwebVersion .~ Development
       , [IncorrectHash, IncorrectPow, VersionMismatch]
       )
     , ( hdr & testHeaderHdr . blockWeight .~ 10
-      , [IncorrectHash, IncorrectPow, IncorrectWeight]
+      , [IncorrectHash, IncorrectPow, IncorrectWeight, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockWeight %~ (+ 1)
-      , [IncorrectHash, IncorrectPow, IncorrectWeight]
+      , [IncorrectHash, IncorrectPow, IncorrectWeight, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockHeight .~ 10
-      , [IncorrectHash, IncorrectPow, IncorrectHeight]
+      , [IncorrectHash, IncorrectPow, IncorrectHeight, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockHeight %~ (+ 1)
-      , [IncorrectHash, IncorrectPow, IncorrectHeight]
+      , [IncorrectHash, IncorrectPow, IncorrectHeight, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockNonce .~ Nonce 0
-      , [IncorrectHash, IncorrectPow]
+      , [IncorrectHash, IncorrectPow, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockNonce %~ Nonce . (+1) . encodeNonceToWord64
-      , [IncorrectHash, IncorrectPow]
+      , [IncorrectHash, IncorrectPow, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockFlags .~ fromJuste (runGet decodeFeatureFlags badFlags)
-      , [IncorrectHash, IncorrectPow]
+      , [IncorrectHash, IncorrectPow, InvalidFeatureFlags]
       )
     , ( hdr & testHeaderHdr . blockAdjacentHashes .~ BlockHashRecord mempty
-      , [IncorrectHash, IncorrectPow]
+      , [IncorrectHash, IncorrectPow, InvalidFeatureFlags]
       )
     ]
```

Comment:
Instead of failing with `InvalidFeatureFlags` (almost) everywhere, wouldn't it be better to fix the test header or replace it with one that has correct feature flags?

There are utils in `Test.Chainweb.Utils.TestHeader` for picking a TestHeader from the history of an existing Chainweb instance.